CMakeLists.txt

```diff
@@ -604,96 +606,100 @@ if(FFMPEG_FOUND)
   )
 endif(FFMPEG_FOUND)
 
-
-foreach(e ${LIBVNCSERVER_EXAMPLES})
-  add_executable(examples_${e} ${LIBVNCSRVEXAMPLE_DIR}/${e}.c)
-  set_target_properties(examples_${e} PROPERTIES OUTPUT_NAME ${e})
-  set_target_properties(examples_${e} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/examples)
-  target_link_libraries(examples_${e} vncserver ${CMAKE_THREAD_LIBS_INIT} ${CARBON_LIBRARY} ${IOKIT_LIBRARY} ${IOSURFACE_LIBRARY})
-endforeach(e ${LIBVNCSERVER_EXAMPLES})
-
-foreach(e ${LIBVNCCLIENT_EXAMPLES})
-  add_executable(client_examples_${e} ${LIBVNCCLIEXAMPLE_DIR}/${e}.c ${LIBVNCCLIEXAMPLE_DIR}/${${e}_EXTRA_SOURCES} )
-  set_target_properties(client_examples_${e} PROPERTIES OUTPUT_NAME ${e})
-  set_target_properties(client_examples_${e} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/client_examples)
-  target_link_libraries(client_examples_${e} vncclient ${CMAKE_THREAD_LIBS_INIT} ${SDL2_LIBRARY} ${GTK2_LIBRARIES} ${FFMPEG_LIBRARIES})
-endforeach(e ${LIBVNCCLIENT_EXAMPLES})
-
+if(WITH_EXAMPLES)
+  foreach(e ${LIBVNCSERVER_EXAMPLES})
+    add_executable(examples_${e} ${LIBVNCSRVEXAMPLE_DIR}/${e}.c)
+    set_target_properties(examples_${e} PROPERTIES OUTPUT_NAME ${e})
+    set_target_properties(examples_${e} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/examples)
+    target_link_libraries(examples_${e} vncserver ${CMAKE_THREAD_LIBS_INIT} ${CARBON_LIBRARY} ${IOKIT_LIBRARY} ${IOSURFACE_LIBRARY})
+  endforeach(e ${LIBVNCSERVER_EXAMPLES})
+
+  foreach(e ${LIBVNCCLIENT_EXAMPLES})
+    add_executable(client_examples_${e} ${LIBVNCCLIEXAMPLE_DIR}/${e}.c ${LIBVNCCLIEXAMPLE_DIR}/${${e}_EXTRA_SOURCES} )
+    set_target_properties(client_examples_${e} PROPERTIES OUTPUT_NAME ${e})
+    set_target_properties(client_examples_${e} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/client_examples)
+    target_link_libraries(client_examples_${e} vncclient ${CMAKE_THREAD_LIBS_INIT} ${SDL2_LIBRARY} ${GTK2_LIBRARIES} ${FFMPEG_LIBRARIES})
+  endforeach(e ${LIBVNCCLIENT_EXAMPLES})
+endif(WITH_EXAMPLES)
 
 #
 # them tests
 #
 
-if(UNIX)
-    set(ADDITIONAL_TEST_LIBS m)
-endif(UNIX)
+if(WITH_TEST)
```

Comment:
that diff looks huge. please only add the `if(WITH_TESTS)` and `endif(WITH_TESTS)` lines. i take respsonsibility for the wrong indent of contained lines ;-)

Reply:
Changed :).